### ISSUE TITLE
Allow deviation for Identities

### ISSUE THREAD

**Issue by BalazsLengyel** [netmod-wg/yang-next#40]
It should be possible to declare that I  do not support specific identities in a module. 

E.g. I want to indicate which interface-types I support from ian-if-types.yang

**Comment by schoenw** [comment]
Yes, but the solution for #40 can't be #41.

**Comment by mbj4668** [comment]
I agree with the problem statement, but I don't think that deviations is the right tool.

**Comment by kwatsen** [comment]
Rob doesn't think deviations would be the way to solve this.
OpenConfig defines their own interface-types, a subset of iana-if-types.

Andy wants to ensure that we strive for a few holistic solutions rather than a bunch of point solutions.
Not sure if a YANG Module is the right place.

PSUEDO Closed during Feb 6 Virtual Interim meeting.

*** An Issue SHOULD be opened against NC/RC to solve this problem ***

**Comment by abierman** [comment]
This is a server capabilities problem.
Static capabilities are being handled today with deviations but existing deviate(not-supported)
statements would be verbose and difficult to write. (e.g., iana-if-type)

An rpc solution would require server code to support, but would allow complex proprietary logic
to be used to determine the allowed identities. 

Now that if-feature is allowed in an identity, the conformance is very clear.
There is no way for a server to express any deviations for an identityref.
Not even replacing the type-stmt will work,

This should be an important problem to solve, not Unknown.
The backward compatibility should be high, since no existing statements
will be altered or removed to support this feature.
Complexity is probably high.

**Comment by kwatsen** [comment]
This was "pseudo-closed" above, should we create NC-next/RC-next issues for it now?

**Comment by lllyfeng** [comment]
allow identity-ref to permit/deny some idenities is another way.
like this:
type identity-ref {
    base iana:if-type {
        permit xxxx;
       deny xxxx;
   }

}

**Comment by rgwilton** [comment]
Also see #80, #107 and #124

**Comment by kwatsen** [comment]
Jan thinks this is more a YANG-issue, not a proto-binding issue (and agrees that deviations are not ideal)

Juergen notes that there is also a runtime-dependency...so both?
